src/state/notifications/sagas.ts

```diff
@@ -111,35 +130,71 @@ export function* unsubscribeWalletSaga(action: UnsubscribeWalletAction) {
 
 export function* authenticateEmailSaga(action: AuthenticateEmailAction) {
   const { payload, meta } = action as AuthenticateEmailAction;
+
   try {
     const response = yield call(authenticateEmail, payload);
+
     if (response.result === Result.success) {
       yield put(authenticateEmailSuccess());
       if (meta?.onSuccess) {
-        meta.onSuccess();
+        meta.onSuccess(payload.session_token);
       }
     }
   } catch (error) {
     const { msg } = error.response.data;
+
     yield put(authenticateEmailFailure(msg));
     if (meta?.onFailure) {
       meta.onFailure();
     }
   }
 }
 
+export function* updateEmailSaga(action: UpdateNotificationEmailAction) {
+  const {
+    payload: { wallets, currentEmail, newEmail },
+    meta,
+  } = action;
+
+  try {
+    const walletsWithHashes = yield Promise.all(
```

Comment:
check subscribeWalletSaga how it is done with the usage of all and call from sagas, we should use sagas way for doing it